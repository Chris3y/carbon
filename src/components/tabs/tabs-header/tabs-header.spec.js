import React from 'react';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import TabsHeader from './tabs-header.component';
import StyledTabsHeader from './tabs-header.style';
import classicTheme from '../../../style/themes/classic';
import TabHeader from '../tab-header/tab-header.component';

function render(props) {
  return shallow(
    <TabsHeader { ...props }>
      <TabHeader title='title-1' tabId='tabId-1' />
      <TabHeader title='title-1' tabId='tabId-1' />
    </TabsHeader>
  );
}

function renderStyles(props) {
  return TestRenderer.create(<StyledTabsHeader { ...props } />);
}

describe('TabsHeader', () => {
  let wrapper;
  it('renders as expected', () => {
    wrapper = renderStyles();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders children correctly if passed', () => {
    wrapper = render();
    expect(wrapper.children()).toHaveLength(2);
  });

  describe('when position prop is set to left', () => {
    it('applies proper styles', () => {
      wrapper = renderStyles({ position: 'left ' });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in classic style', () => {
    it('renders as expected', () => {
      wrapper = renderStyles({ theme: classicTheme });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
