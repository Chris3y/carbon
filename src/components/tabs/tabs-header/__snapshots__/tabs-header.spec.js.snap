// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsHeader renders as expected 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: inset 0px -2px 0px 0px #E5EAEC;
  cursor: pointer;
  list-style: none;
  margin: 0 0 10px;
  padding: 0;
}

<ul
  className="c0"
/>
`;

exports[`TabsHeader when align prop is set to right applies proper styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: inset 0px -2px 0px 0px #E5EAEC;
  cursor: pointer;
  list-style: none;
  margin: 0 0 10px;
  padding: 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  text-align: right;
}

<ul
  className="c0"
/>
`;

exports[`TabsHeader when in classic style renders as expected 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: inset 0px -2px 0px 0px #E5EAEC;
  cursor: pointer;
  list-style: none;
  margin: 0 0 10px;
  padding: 0;
  box-shadow: inset 0px -2px 0px 0px #ccd6db;
}

<ul
  className="c0"
/>
`;

exports[`TabsHeader when position prop is set to left applies proper styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: inset 0px -2px 0px 0px #E5EAEC;
  cursor: pointer;
  list-style: none;
  margin: 0 0 10px;
  padding: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: inset -2px 0px 0px 0px #E5EAEC;
  width: 20%;
  margin: 0 10px 0;
}

<ul
  className="c0"
/>
`;
